bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/action/ScriptExecution.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.automation.module.script.action;
+
+import java.time.ZonedDateTime;
+import java.util.function.Consumer;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * The {@link ScriptExecution} allows creating timers for asynchronous script execution
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+public interface ScriptExecution {
+
+    /**
+     * Schedules a block of code for later execution.
+     *
+     * @param instant the point in time when the code should be executed
+     * @param closure the code block to execute
+     * @return a handle to the created timer, so that it can be canceled or rescheduled
+     */
+    Timer createTimer(ZonedDateTime instant, Runnable closure);
```

Comment:
The naming of the `ZonedDateTime` as _instant_ triggers me to ask the question: Would an `Instant` be sufficient here, since the the point in time doesn't care about time-zone? If `ZonedDateTime` is more convenient, consider if parameter should be renamed for clarity, and/or if a variant taking `Instant` would be relevant.

Reply:
The docs actually say it should be an `AbstractInstant`.

Reply:
`AbstractInstant` is surely wrong because that's a class from soda-time which we don't use anymore. So the docs need to be updated.

IMO an `Instant` would also be fine (also see my comment on #2898) but I re-implemented what we have in DSL and that is a `ZonedDateTime`. If we try to solve that now, we won't get a solution before next release.

Reply:
Out of interest: Why can't we get a solution before next release?

Reply:
I always felt that the docs references to abstract classes like this was confusing to new users anyway. I'm not sure `Instant` is any better but it's at least consistent. When/if this ever changes here to `Instant`, we should also change the persistence actions which are documented as using `ZonedDateTime` to be consistent.

Reply:
@florian-h05 Technically it might be possible, but it's important what @rkoshak mentions: We use `ZonedDateTime` in many places. So we should have a discussion where to go, having `Instant` for `createTimer` and `ZonedDateTime` for the persistence services is very confusing and should be avoided. 

General discussions like this tend to take some time and with less than a month until feature-freeze and then time maybe a bit short. Just "exchanging" it with `Instant` will probably break a lot of code and I feel it's not a good idea to do that for a minor release.

Reply:
Thanks for the explanation, I understand the impact of such a decision. 

So what’s the plan now? Get this PR merged before the next release and then discuss the Instand ZonedDateTime thing later?

I’d like to get the thread-safe reimplementation of createTimer in JS Scripting into the next release.

Reply:
That would be my proposal. IMO user-facing ZDT and Instant don‘t make much difference, so using Instant internally in DateTimeType and ZDT in rules is fine. Does it make a difference in JS?

Reply:
> Does it make a difference in JS?

Regarding the compatibility, no.
We are using JS-Joda as date and time library, which provides the same functionality that the Joda time library had (ZonedDateTime, Instant and so on).
When core moves from ZonedDateTime to Instant, we only need to use Instant from JS-Joda instead of ZonedDateTime from JS-Joda.

Reply:
If there is no real benefit (I don't see one for DSL or JavaRule), we should probably stay where we are.

Reply:
The only difference I can see is, which argument and which string the `parse` function takes.
Given this little difference, but the large number of scripts that have to be updated when that’s changed to an Instant, I agree that we should probably stay on ZonedDateTime.
The question for now is: should we rename the parameter from instant to zdt or zonedDateTime or something like that?

Reply:
Yes, I‘ll update that.